autoload/buffer.vim

```diff
@@ -259,7 +259,12 @@ def CurBufMatches(prefix: string): list<dict<any>>
     return candidates
 enddef
 
+var _last_base_len = 666
```

Comment:
What problem is _last_base_len trying to solve?

Why combine 2 unrelated changes in one PR?